### PR TITLE
Small maintenance

### PR DESCRIPTION
**Pull request by onur9**
Some small improvements in the build pipeline: added internet-recommended build flags and couple of new goreleaser steps to improve packaging. New linting checks + improvements in the code organization (in accordance with https://github.com/golang-standards/project-layout)

### Test pipeline

- Renamed to linux to reflect the tested platform name in the actions UI.
- Do not output artifacts for all platforms in the test pipeline, only run `make run`.
- Make test pipeline run on pull_requests as well. This should be cheap without the snapshots build.
- Use 1.18.x as golang version.
- Use the official setup-go library.
- Set up go modules cache.
- Replace staticcheck step with `make lint cover errcheck` in the test pipeline
- Removed all other settings from checkout code step except for fetch-depth.
- ~~Replaced Upload to codecov step with Upload to Github actions artifact.~~ Next to codecov step, github outputted artifacts are: Binary for the tested platform (linux-amd64) and the `coverage.out` file, which can be turned into HTML locally using `make cover-html`

### Release pipeline
- Run tests with `make pedantic`
- Use 1.18.x as golang version
- Use the official setup-go library.
- Set up go modules cache.

### Goreleaser

- Added before hook, runs `make tidy`
- Since this PR also adds the `cmd` directory, I changed the build config to work with multiple binaries instead of one.
- Added some recommended build options to enforce statically linked binaries.
- Previously goreleaser passed build-time args to `utils` module, this change passes them to the `main` instead.
- Added archive step, this creates a nice tar.gz file on the Releases page.
- Added milestones step, although not used currently, this is used for closing milestones if any based on the version number.
- ~~Added signs step for signing the output.~~ (Commented out until we add fingerprint and key in repo settings)

### Makefile

- Added new linting checks.
- Added build configuration for creating dev builds.
- Works with multiple binaries as well.

### File organization

- The contents of `cmd` are moved to `cmd/mach-composer`
- All internal libraries are moved to `internal/` folder.

### Linting fixes

- Some error handling.
- Use standard go naming.
- Formatting.

### CODE REVIEW

File: internal/generator/filters.go [hunk 1]
```diff
@@ -3,27 +3,51 @@ package generator
...
-	pongo2.RegisterFilter("tfvalue", filterTFValue)
-	pongo2.RegisterFilter("render_tf_provider", renderTFProvider)
-	pongo2.RegisterFilter("azure_frontend_endpoint_name", AzureFrontendEndpointName)
-	pongo2.RegisterFilter("service_plan_resource_name", AzureServicePlanResourceName)
-	pongo2.RegisterFilter("get", FilterGetValueByKey)
-	pongo2.RegisterFilter("render_commercetools_scopes", filterCommercetoolsScopes)
-	pongo2.RegisterFilter("component_endpoint_name", filterComponentEndpointName)
+	if err := pongo2.RegisterFilter("string", filterString); err != nil {
+		panic(fmt.Errorf("pongo2.RegisterFilter('string'): %v", err))
+	}
```
Comment (mvantellingen) [inline internal/generator/filters.go:17]: Perhaps we can better use `pongo2.Must(pongo2.RegisterFilter(...))`

**Comment by mvantellingen** [comment]
Looks good! Two minor comments added.

File: internal/generator/filters.go (hunk 1 above)
Comment (onur9) [reply internal/generator/filters.go:17]: Is it `MustRegisterFilter`? Not sure, what this library does but `Must` seems to be something else.

Comment (onur9) [reply internal/generator/filters.go]: Added a `mustRegisterFilter()` function for this.

File: cmd/mach-composer/version.go [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
 	RunE: func(cmd *cobra.Command, args []string) error {
-		fmt.Printf("mach-composer %s (%s) - %s\n", utils.Version, utils.Commit[:7], utils.Date)
+		if commit != "" && date != "" {
+			fmt.Printf("v%s (%s) - %s\n", version, commit[:7], date)
+		} else {
+			if version != "development" {
+				version = fmt.Sprintf("v%s", version)
+			}
+			fmt.Println(version)
+		}
```
[the remark this answers is not in the document]
Comment (onur9) [reply cmd/mach-composer/version.go:15]: It is reverted.

**Pull request merged**